apps/email/js/sync.js

```diff
@@ -311,17 +310,23 @@ define(function(require) {
             }
 
             if (model.getAccountCount() === 1) {
-              subject = mozL10n.get('new-emails-notify-one-account', {
-                n: count
-              });
+              subjectL10n = { id: 'new-emails-notify-one-account',
+                              args: { n: count }
+                            };
             } else {
-              subject = mozL10n.get('new-emails-notify-multiple-accounts', {
-                n: count,
-                accountName: result.address
-              });
+              subjectL10n = { id: 'new-emails-notify-multiple-accounts',
+                              args: {
+                                n: count,
+                                accountName: result.address
+                              }
+                            };
             }
+            
+            mozL10n.formatValue('senders-separation-sign')
+              .then(separator => {
+                bodyL10n = newFromNames.join(separator);
+            });
```

Comment:
you can't just go on with your code after async `formatValue`. Everything behind this code that is used until return has to be within the Promise then callback.